distros/build-ros-package/default.nix

```diff
@@ -28,4 +30,35 @@ else stdenv.mkDerivation) (args // {
   };
 } // lib.optionalAttrs (buildType == "ament_python") {
   dontUseCmakeConfigure = true;
+
+  # Modelled after colcon:
+  # https://github.com/colcon/colcon-core/blob/master/colcon_core/task/python/build.py
```

Comment:
More comments would be useful. I guess that sooner or later, colcon will be updated not to use the deprecated install methods and then somebody will have to figure out what was changes and update the code below accordingly. So we should mention at least the colcon commit after which it is modeled.

We could also mention the important code is in the `build` method and that we follow the `if not args.symlink_install or 'develop' not in available_commands` branch in it.